Secure-Software-Development-Policy/Zowe-SSDP-Organization/Policy.md

```diff
@@ -0,0 +1,206 @@
+
+```#REF: SSFD:PO Prepare the organization```
+
+#### A1. Roles and Responsibilities (#ID: ZSSD-OP:PO-RR)
+```#REF: SSDF:PO.2 Implement Roles and Responsibilities```
+```#DISCARD```
+
+TODO: Consider skipping this in full as it probably makes no sense to require individual contributors to get commitment and roles and trainings.
+
+- A1.1 Create Roles and Alter responsibilities (#ID: ZSSD-OP:PO-RR-CRA) 
+
+Create new roles and alter responsibilities for existing roles as needed to encompass all parts of the SSDF. 
+Periodically review and maintain the defined roles and responsibilities, updating them as needed\
+
+```#REF: SSDF:PO.2.1```
+
+- A1.2 Role Based Training (#ID: ZSSD-OP:PO-RR-RBT)
+  
+Provide role-based training for all personnel with responsibilities that contribute to secure development. 
+Periodically review personnel proficiency and role-based training, and update the training as needed.
+
+```#REF: SSDF:PO.2.2, SSF-A.1.1-B, SSF-A.1.2-B```\
+```#NOT-APPLICABLE```
+
+- A1.3 Upper Management Commitment (#ID: ZSSD-OP:PO-RR-UMC)
+
+Obtain upper management commitment to secure development, and convey that commitment to all with SSDF related roles and responsibilities.
+
+```#REF: SSDF:PO.2.3```\
+```#NOT-APPLICABLE for individuals and small organizations.```\
+```#HARD-TO-PROVE```
+
+#### A2. Supporting Toolchains (#ID: ZSSD-OP:PO-ST)
+```#REF: SSFD:PO.3 Implement Supporting Toolchains```\
+```#KEEP #RE-DEFINE```
+
+- A2.1 Toolchain Components Integration (#ID: ZSSD-OP:PO-ST-TCI)
+```#REF: SSDF:PO.3.1```
+```#KEEP```
+
+Specify which tools or tool types must or should be included in each toolchain to mitigate identified risks, as well as how the toolchain components are to be integrated with each other
+
+```TODO: The organization/contributors developing Zowe component, should either reuse the tools maintained by Zowe or shall use own tools adequate to the once provided by Zowe and providing consistent results to satisfy security criteria defined in both ZSSD policies.```     
+
+- A2.2:Deploy and Maintain Tools (#ID: ZSSD-OP:PO-ST-DMT)\
+```#REF: SSDF:PO.3.2```
+```#KEEP```
+
+Follow recommended security practices deploying and maintaining tools and toolchains.
+
+```TODO: consider removing or mergin with the previous one or be flexible enough to have an option for different kinds of contributors :individual, squad, company with own tool chains?```
+
+- A2.3 Generate Evidence and Artifacts (#ID: ZSSD-OP:PO-ST-GEA)\
+```#REF: SSDF:PO.3.3```\
+```#KEEP```
+
+Configure tools to generate evidence and artifacts of their support of secure software development practices as defined by Zowe.
+
+#### A3. Security Criteria (#ID: ZSSD-OP:PO-SC) 
+```#REF:  SSFD:PO.4 Define and Use Criteria for Software Security Checks)```\
+```#KEEP```
+
+Provide some kind of automatic validation for the Zowe Secure Development criteria - validate during pipeline execution.
```

Comment:
To maintain automation, i think we would need to have a standard set of tools whose results we accept. This standard set would allow automated parsing of things like security scan results (check for # of high vulns, total number overall, etc). Maybe we should look into free, open source code scanners like VisualCodeGrepper to make requirements on developers easier to achieve?

Reply:
This tool is allegedly free: https://community.synopsys.com/s/topic/0TO34000000M9NjGAK/download-coverity-analysis

Reply:
For SAST we usually use SonarCloud at the moment.